experiments/TestCase/risingbubble.jl

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env julia --project
+using ArgParse
+
+using ClimateMachine
+using ClimateMachine.Atmos
+using ClimateMachine.Orientations
+using ClimateMachine.ConfigTypes
+using ClimateMachine.Diagnostics
+using ClimateMachine.GenericCallbacks
+using ClimateMachine.ODESolvers
+using ClimateMachine.TemperatureProfiles
+using ClimateMachine.Thermodynamics
+using ClimateMachine.TurbulenceClosures
+using ClimateMachine.VariableTemplates
+using StaticArrays
+using Test
+using CLIMAParameters
+using CLIMAParameters.Atmos.SubgridScale: C_smag
+using CLIMAParameters.Planet: R_d, cp_d, cv_d, MSLP, grav
+struct EarthParameterSet <: AbstractEarthParameterSet end
+const param_set = EarthParameterSet();
+
+function init_risingbubble!(problem, bl, state, aux, (x, y, z), t)
+    ## Problem float-type
+    FT = eltype(state)
+
+    ## Unpack constant parameters
+    R_gas::FT = R_d(bl.param_set)
+    c_p::FT = cp_d(bl.param_set)
+    c_v::FT = cv_d(bl.param_set)
+    p0::FT = MSLP(bl.param_set)
+    _grav::FT = grav(bl.param_set)
+    γ::FT = c_p / c_v
+
+    ## Define bubble center and background potential temperature
+    xc::FT = 5000
+    yc::FT = 1000
+    zc::FT = 2000
+    r = sqrt((x - xc)^2 + (z - zc)^2)
+    rc::FT = 2000
+    θamplitude::FT = 2
+    q_tot_amplitude::FT = 1e-3
+
+    ## TODO: clean this up, or add convenience function:
+    ## This is configured in the reference hydrostatic state
+    θ_ref::FT = bl.ref_state.virtual_temperature_profile.T_surface
+
+    ## Add the thermal perturbation:
+    Δθ::FT = 0
+    Δq_tot::FT = 0
+    if r <= rc
+        Δθ = θamplitude * (1.0 - r / rc)
+        Δq_tot = q_tot_amplitude * (1.0 - r / rc)
+    end
+
+    ## Compute perturbed thermodynamic state:
+    θ = θ_ref + Δθ                                      # potential temperature
+    q_pt = PhasePartition(Δq_tot)
+    R_m = gas_constant_air(bl.param_set, q_pt)
+    _cp_m = cp_m(bl.param_set, q_pt)
+    _cv_m = cv_m(bl.param_set, q_pt)
+    π_exner = FT(1) - _grav / (_cp_m * θ) * z           # exner pressure
+    ρ = p0 / (R_gas * θ) * (π_exner)^(_cv_m / R_m)      # density
+    T = θ * π_exner
+
+    if bl.moisture isa EquilMoist
```

Comment:
Should this work even with `NonEquilMoist`? If so, maybe testing for `DryModel` is better.

Reply:
I'm not sure whether this works with `NonEquilMoist`, but I think it's better to test for `DryModel`. I will change it.